back-end/java/main/java/src/DormRoom/DormBuilding.java

```diff
@@ -0,0 +1,192 @@
+package src.DormRoom;
+
+/**
+ * The {@code DormBuilding} class represents data about a specific dormitory building on campus.
+ * Each instance contains information such as the building's name, its location (latitude
+ * and longitude), the year it was built, and its corresponding campus area. Additionally,
+ * it captures the ratio of the building's occupancy to the number of available washing machines
+ * (peoplePerWasher), which provides insight into the building's laundry capacity.
+ *
+ * <p>Constructors create an object fully initialized with the building's attributes,
+ * inferred from the building name. This approach ensures that instances of {@code DormBuilding}
+ * remain consistent and standardized across the application.</p>
+ */
+public class DormBuilding {
+
+  /**
+   * The name of the dorm building. Examples include "Barbour Hall", "Buxton House", and
+   * "Grad Center A".
+   */
+  final String buildingName;
+  /**
+   * The ratio of total building occupancy to the number of available washing machines.
+   * A higher value indicates more people sharing a single washer, while a lower value
+   * suggests better laundry accessibility for residents.
+   */
+  final float peoplePerWasher;
+  /**
+   * The year in which the building was constructed. This can be useful for historical data,
+   * maintenance schedules, or evaluating the age of infrastructure.
+   */
+  final int year;
+  /**
+   * The longitude coordinate of the building's location on campus, given in decimal degrees.
+   * This can be used in geographical calculations, mapping features, or integrated with
+   * location-based services.
+   */
+  final double longitude;
+  /**
+   * The latitude coordinate of the building's location on campus, given in decimal degrees.
+   * Used in conjunction with longitude, this allows the building's position to be plotted
+   * or integrated with GIS systems.
+   */
+  final double latitude;
+  /**
+   * The general location or neighborhood of the building on the campus. This can be used
+   * for wayfinding, categorizing accommodations, or filtering buildings by campus area.
+   */
+  final CampusLocation campusLocation;
+
+
+  /**
+   * Constructs a new {@code DormBuilding} using the supplied building name to determine
+   * its attributes. This constructor uses a predefined mapping of building names to their
+   * associated details (e.g., geographical coordinates, construction year, and laundry ratio).
+   *
+   * @param buildingName the official name of the building (e.g. "Barbour Hall", "Buxton House")
+   * @throws RuntimeException if the building name does not match any known buildings in the predefined mapping
+   */
+  public DormBuilding(String buildingName) {
```

Comment:
I feel like there's a better way to do this but honestly there probably isn't You can ignore this comment just something to think about

Reply:
Discussed in conversation and chosen to continue with current switch implementation